packages/sync/src/utils/initSyncComponents.ts

```diff
@@ -0,0 +1,94 @@
+import { NON_SYNCING_TABLES } from '../constants';
+
+interface Row {
+  [key: string]: string | number | boolean | null;
+}
+interface DatabaseDriver {
+  runSql: (sql: string) => Promise<{ rows: Row[] }>;
+}
+
+const TABLES_WITHOUT_COLUMN_QUERY = `
+  SELECT
+    pg_class.relname as table
+  FROM
+    pg_catalog.pg_class
+  JOIN
+    pg_catalog.pg_namespace
+  ON
+    pg_class.relnamespace = pg_namespace.oid
+  LEFT JOIN
+    pg_catalog.pg_attribute
+  ON
+    pg_attribute.attrelid = pg_class.oid
+  AND
+    pg_attribute.attname = 'updated_at_sync_tick'
+  WHERE
+    pg_namespace.nspname = 'public'
+  AND
+    pg_class.relkind = 'r'
+  AND
+    pg_attribute.attname IS NULL
+  AND
+    pg_class.relname NOT IN (${NON_SYNCING_TABLES.map(t => `'${t}'`).join(',')});
+`;
+
+const TABLES_WITHOUT_TRIGGER_QUERY = `
+  SELECT
+    t.table_name as table
+  FROM
+    information_schema.tables t
+  LEFT JOIN
+    information_schema.table_privileges privileges
+  ON
+    t.table_name = privileges.table_name AND privileges.table_schema = 'public'
+  WHERE
+    NOT EXISTS (
+      SELECT
+        *
+      FROM
+        pg_trigger p
+      WHERE
+        p.tgname = substring(concat('set_', lower(t.table_name), '_updated_at_sync_tick'), 0, 64)
+    )
+  AND
+    privileges.privilege_type = 'TRIGGER'
+  AND
+    t.table_schema = 'public'
+  AND
+    t.table_type != 'VIEW'
+  AND
+    t.table_name NOT IN (${NON_SYNCING_TABLES.map(t => `'${t}'`).join(',')});
+`;
+
+// TODO: fix type for driver
+export async function initSyncComponents(driver: DatabaseDriver, isClient: boolean = false) {
+  // add column: holds last update tick, default to -999 (not marked for sync) on client,
```

Comment:
```suggestion
  // add column: holds last update tick, default to -999 (not modified locally) on client,
```
We need to be careful about "marked for sync", as it generally means "marked for _ongoing_ sync" in the Tamanu context, i.e. a patient is marked as being a patient of interest to that facility. This is a different concept, where the record is seen as modified locally, aka "dirty"